Remove RTCStats default constructors

youennf opened the pull request:
#### 6311892c70d834b4aede4a04d4ce4ccc136063f4
<pre>
Remove RTCStats default constructors
<a href="https://rdar.apple.com/129206906">rdar://129206906</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=275098">https://bugs.webkit.org/show_bug.cgi?id=275098</a>

Reviewed by Sam Weinig.

We use JSGenerateToNativeObject to enable convertToDictionary explicitly.
It is also generated if the dictionary does not have the JSGenerateToJSObject keyword, as often dictionaries are either in or out.
This patch add both keywords for in/out dictionaries.

We can then remove RTC stats default constructors since these are only out dictionaries.

Canonical link: <a href="https://commits.webkit.org/279854@main">https://commits.webkit.org/279854@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9f17cac09c760c6ecc8f9f358ee25c1a443a4abb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54470 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/33883 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/7034 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/57749 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5201 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/56773 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/41417 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/5191 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/44108 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3489 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/56565 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32020 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47173 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25244 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/28818 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/4502 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3344 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/50630 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/4716 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59340 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/29696 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/4870 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/51532 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/30849 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47262 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/50902 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12014 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/31828 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/30641 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

weinig reviewed (approved):
I went a different direction in my WIP PR https://github.com/WebKit/WebKit/pull/29056/files, and added a SuppressToNative to do the reverse, but I like the explicitness of this approach.

youennf committed:
Remove RTCStats default constructors
rdar://129206906
https://bugs.webkit.org/show_bug.cgi?id=275098

Reviewed by Sam Weinig.

We use JSGenerateToNativeObject to enable convertToDictionary explicitly.
It is also generated if the dictionary does not have the JSGenerateToJSObject keyword, as often dictionaries are either in or out.
This patch add both keywords for in/out dictionaries.

We can then remove RTC stats default constructors since these are only out dictionaries.

Canonical link: https://commits.webkit.org/279854@main
